include/ddc/kernels/splines/spline_builder.hpp

```diff
@@ -559,68 +559,122 @@ operator()(
         assert(derivs_xmin->template extent<deriv_type>() == s_nbc_xmin);
         auto derivs_xmin_values = *derivs_xmin;
         auto const dx_proxy = m_dx;
-        ddc::for_each(
-                ddc::policies::policy(exec_space()),
-                batch_domain(),
-                KOKKOS_LAMBDA(typename batch_domain_type::discrete_element_type j) {
-                    for (int i = s_nbc_xmin; i > 0; --i) {
-                        spline(ddc::DiscreteElement<bsplines_type>(s_nbc_xmin - i), j)
-                                = derivs_xmin_values(ddc::DiscreteElement<deriv_type>(i), j)
-                                  * ddc::detail::ipow(dx_proxy, i + s_odd - 1);
-                    }
-                });
+        if constexpr (std::is_same<exec_space, Kokkos::Serial>::value) {
+            ddc::for_each(
+                    batch_domain(),
+                    KOKKOS_LAMBDA(typename batch_domain_type::discrete_element_type j) {
+                        for (int i = s_nbc_xmin; i > 0; --i) {
+                            spline(ddc::DiscreteElement<bsplines_type>(s_nbc_xmin - i), j)
+                                    = derivs_xmin_values(ddc::DiscreteElement<deriv_type>(i), j)
+                                      * ddc::detail::ipow(dx_proxy, i + s_odd - 1);
+                        }
+                    });
+        } else {
+            ddc::parallel_for_each<exec_space>(
+                    exec_space(),
+                    batch_domain(),
+                    KOKKOS_LAMBDA(typename batch_domain_type::discrete_element_type j) {
+                        for (int i = s_nbc_xmin; i > 0; --i) {
+                            spline(ddc::DiscreteElement<bsplines_type>(s_nbc_xmin - i), j)
+                                    = derivs_xmin_values(ddc::DiscreteElement<deriv_type>(i), j)
+                                      * ddc::detail::ipow(dx_proxy, i + s_odd - 1);
+                        }
+                    });
+        }
```

Comment:
You can also apply this to all the rest
```suggestion
            ddc::parallel_for_each(
                    exec_space(),
                    batch_domain(),
                    KOKKOS_LAMBDA(typename batch_domain_type::discrete_element_type j) {
                        for (int i = s_nbc_xmin; i > 0; --i) {
                            spline(ddc::DiscreteElement<bsplines_type>(s_nbc_xmin - i), j)
                                    = derivs_xmin_values(ddc::DiscreteElement<deriv_type>(i), j)
                                      * ddc::detail::ipow(dx_proxy, i + s_odd - 1);
                        }
                    });
```